### PR TITLE
Add developer guide

### PR DESCRIPTION
**Pull request by tuteng**
### Modifications
* Add user developer guider

### CODE REVIEW

File: docs/developer-guide.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+* Mybatis
+* PageHelper
+* Swagger2
+* Zuul
+
+The code structure of the backend is shown below.
+
+![backend-code](img/pulsar-manager-backend-code.png)
+
+The backend configuration file is [here](https://github.com/streamnative/pulsar-manager/blob/master/src/main/resources/application.properties).
```
Comment (Anonymitaet) [inline docs/developer-guide.md]:
```suggestion
* The backend configuration file is [here](https://github.com/streamnative/pulsar-manager/blob/master/src/main/resources/application.properties).
```

File: docs/developer-guide.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+The database schema is defined [here](https://github.com/streamnative/pulsar-manager/blob/master/src/main/resources/META-INF/sql/mysql-schema.sql)
```
Comment (Anonymitaet) [inline docs/developer-guide.md]:
```suggestion
* The database schema is defined [here](https://github.com/streamnative/pulsar-manager/blob/master/src/main/resources/META-INF/sql/mysql-schema.sql)
```

File: docs/developer-guide.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+Add new RESTful api
```
Comment (Anonymitaet) [inline docs/developer-guide.md]:
```suggestion
* Add new RESTful API
```

File: docs/developer-guide.md (hunk 1 above)
Comment (Anonymitaet) [reply docs/developer-guide.md]: Currently, it looks like:  

image

File: docs/developer-guide.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+Restful api defines the specification. If the RESTful api needs backend processing, it needs to be prefixed with pulsar-manager. If you want to directly forward the request to the broker, you can directly define it according to the specification of Pulsar RESTful api.
+
+Let's take a new tenant as an example to illustrate how to develop in pulsar manager.
```
Comment (Anonymitaet) [inline docs/developer-guide.md]:
```suggestion
Let's take a new tenant as an example to illustrate how to develop in the Pulsar Manager project.
```

File: docs/developer-guide.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+1. Define a RESTful api
+Restful api that requires backend processing
+List all tenants
+GET /pulsar-manager/admin/v2/tenants
+     (2) Restful api directly forwarded to a broker
+Add a new tenant
+PUT /admin/v2/tenants/{tenantName}
+
```
Comment (Anonymitaet) [inline docs/developer-guide.md]: The format is incorrect, please adjust it.

image

File: docs/developer-guide.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+2. Define a service interface.
+Service is used to implement business logic and add interface.
+Here, the `TenantsService` interface is added under the service folder.
+
+![tenants-service](img/tenants-service.png)
+
+3. Implement a service interface.
+Here, some aggregation is done for the data queried from broker.
```
Comment (Anonymitaet) [inline docs/developer-guide.md]:
```suggestion
Here, aggregate data queried from brokers.
```

File: docs/developer-guide.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
```
Comment (Anonymitaet) [inline docs/developer-guide.md]: Please set the contents in the red box to new paragraphs to make things more clear.

image

image

File: docs/developer-guide.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+![tenants-service-impl](img/tenants-service-impl.png)
+
+4. Add a test a test under the test folder, mock data and verify results.
```
Comment (Anonymitaet) [inline docs/developer-guide.md]:
```suggestion
4. Add a test under the test folder, mock data and verify results.
```

File: docs/developer-guide.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+![tenants-service-impl-test](img/tenants-service-impl-test.png)
+
+5. Develop RESTful features.
+Add a `TenantsController` file under the controller folder to add an api to obtain the tenant list.
```
Comment (Anonymitaet) [inline docs/developer-guide.md]:
```suggestion
Add a `TenantsController` file under the controller folder to add an API to obtain the tenant list.
```

File: docs/developer-guide.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+In the following class you can add some description information, error response, etc.
```
Comment (Anonymitaet) [inline docs/developer-guide.md]:
```suggestion
In the following class, you can add some descriptions, error responses, and so on.
```

File: docs/developer-guide.md [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+
```
Comment (Anonymitaet) [inline docs/developer-guide.md]: same layout issue:

image

File: docs/developer-guide.md [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+![tenants-controller](img/tenants-controller.png)
+
+## Frontend
+
+The frontend is developed based on [vue-element-admi](https://panjiachen.github.io/vue-element-admin-site/guide/) and [Element](https://element.eleme.cn/#/en-US).
+The development of the frontend is [here](https://github.com/streamnative/pulsar-manager/tree/master/front-end/src/views/management).
+
```
Comment (Anonymitaet) [inline docs/developer-guide.md]: move line 65-69 here

File: docs/developer-guide.md [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+![frontend-code](img/frontend-code.png)
+
+The frontend code structure is shown as below.
+* api: code of RESTful API icons: code of icons
```
Comment (Anonymitaet) [inline docs/developer-guide.md]:
```suggestion
* api: code of RESTful API icons
```

File: docs/developer-guide.md [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+* router: code of frontend routings
+* lang: code of multiple languages
+* views: code of front pages
+
+### Develop a page
+
+1. Add a RESTful API interface.
+The RESTful interface is defined below the folder api.
```
Comment (Anonymitaet) [inline docs/developer-guide.md]: The RESTful interface is defined below the folder api.

what's the meaning of "is defined below the folder api"?

File: docs/developer-guide.md [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+
+![tenants-api](img/tenants-api.png)
+
+
+2. Add a new page.
+Add a new page [here](https://github.com/streamnative/pulsar-manager/tree/master/front-end/src/views/management). 
+For example, [this](https://github.com/streamnative/pulsar-manager/blob/master/front-end/src/views/management/tenants/index.vue) is a tenant page. )
+
+3. Localization
+You can add  multiple languages to the corresponding configuration folder under the `lang` folder.
```
Comment (Anonymitaet) [inline docs/developer-guide.md]: the same layout issue

image

File: docs/developer-guide.md (hunk 14 above)
Comment (tuteng) [reply docs/developer-guide.md]: https://github.com/streamnative/pulsar-manager/tree/master/front-end/src/api Restful API folder at the front end.

File: docs/developer-guide.md [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+![tenants-controller](img/tenants-controller.png)
+
+## Frontend
+
+The frontend is developed based on [vue-element-admi](https://panjiachen.github.io/vue-element-admin-site/guide/) and [Element](https://element.eleme.cn/#/en-US).
+The development of the frontend is [here](https://github.com/streamnative/pulsar-manager/tree/master/front-end/src/views/management).
+
+![frontend-code](img/frontend-code.png)
+
+The frontend code structure is shown as below.
```
Comment (Anonymitaet) [inline docs/developer-guide.md]:
```suggestion
The frontend code structure is shown as above.
```

Does this sentence explain the image above?

File: docs/developer-guide.md [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+* api: code of RESTful API icons
+* router: code of frontend routings
+* lang: code of multiple languages
+* views: code of front pages
+
+### Develop a page
+
+1. Add a RESTful API interface.
+
+The RESTful interface is defined below the folder api.
```
Comment (Anonymitaet) [inline docs/developer-guide.md]:
```suggestion
The RESTful interface is defined in the `api` folder. 
```
Do you mean this?

File: docs/developer-guide.md (hunk 15 above)
Comment (Anonymitaet) [reply docs/developer-guide.md]: Any thoughts on this layout issue?

**Pull request merged**